src/pythermalcomfort/jos3/thermoregulation.py

```diff
@@ -0,0 +1,1434 @@
+# -*- coding: utf-8 -*-
+
+"""
+This code includes some functions of jos-3 model to calculate human thermoregulation.
+"""
+
+import numpy as np
+import math
+
+from pythermalcomfort.jos3.matrix import NUM_NODES, IDICT, BODY_NAMES
+from pythermalcomfort.jos3 import construction as cons
+
+
+_BSAst = np.array(
```

Comment:
Right. I will at least add a comment about which body part is reffered to each element. When we were writing the original code of JOS3, we hesiteted to use a dictionary because I feel Python is not good at iteration calculation (for loop) and it takes time to calculate. However, using numpy may solve this problem (reference: https://numpy.org/devdocs/user/absolute_beginners.html), we haven't checked the calculation speed though. Let's discuss this later.

Reply:
I agree with you that we should use Numpy to perform the calculations. What we could do is to allow the user to pass a dictionary. We then convert the dictionary to a Numpy array and perform the calculations. This will not negatively affect the speed of the code while at the same time it will ensure that the user enters the right inputs.

We could also use @dataclasses instead of dictionaries to make sure the code is even more robust. Below a simple example.

```
from dataclasses import dataclass

@dataclass
class BodyParts:
    arm: float
    head: float
```

Reply:
I have changed the code to allow the users to pass a dictionary as well as interger value, list, numpy array data format.